pandas/core/generic.py

```diff
@@ -7810,6 +7810,86 @@ def mask(self, cond, other=np.nan, inplace=False, axis=None, level=None,
         is not realigned. That is, use freq if you would like to extend the
         index when shifting and preserve the original data.
 
+        Examples
+        --------
+        Compute the difference between a column in a dataframe
+        with grouped data, and its shifted version.
+
+        >>> data = pd.DataFrame({'myvalue': [1, 2, 3, 4, 5, 6],
+        ...                      'group': ['A', 'A', 'A', 'B', 'B', 'B']},
+        ...                     index=pd.DatetimeIndex(['2016-06-06',
+        ...                                             '2016-06-08',
+        ...                                             '2016-06-09',
+        ...                                             '2016-06-10',
+        ...                                             '2016-06-12',
+        ...                                             '2016-06-13'],
+        ...                                            name='mydate'))
+
+        >>> data
+                    myvalue group
+        mydate
+        2016-06-06        1     A
+        2016-06-08        2     A
+        2016-06-09        3     A
+        2016-06-10        4     B
+        2016-06-12        5     B
+        2016-06-13        6     B
+
+        For the groups compute the difference between current `myvalue` and
+        `myvalue` shifted forward by 1 day.
+
+        If the dataframe is shifted without passing a freq argument than the
+        values simply move down
+
+        >>> data[data.group=='A'].myvalue.shift(1)
```

Comment:
Instead of filtering to group `A` you'd be better served to work with the entire frame

Reply:
Agreed. The group stuff seems to be a distraction from `shift` at this point.